lib/rubocop/cop/rspec/aggregate_examples.rb

```diff
@@ -0,0 +1,294 @@
+                     else
+                       body.source
+                     end
+          "#{base_indent}  #{new_body}"
+        end
+
+        def transform_its(body, arguments)
+          property = arguments.children.first
+          body.source.gsub(/is_expected|are_expected/,
+                           "expect(subject.#{property})")
+        end
+
+        def message_for(examples)
+          if examples.any? { |example| example_with_side_effects?(example) }
+            MSG_FOR_EXPECTATIONS_WITH_SIDE_EFFECTS
+          else
+            MSG
+          end
+        end
+
+        def example_method?(method_name)
+          %i[it specify example scenario].include?(method_name)
+        end
+
+        # Checks if the example:
+        # - exclusively contains expectation statements
+        # - has no metadata (e.g. `freeze: :today`)
+        def_node_matcher :example_without_metadata_and_expectations_solely?,
+                         <<-PATTERN
+          [#example_with_expectations_solely? !#has_metadata?]
+        PATTERN
+
+        def_node_matcher :example_with_expectations_solely?, <<-PATTERN
+          (block #example_block? _
+            { #single_expectation? #all_expectations? }
+          )
+        PATTERN
+
+        # Checks if an example has metadata (e.g. `freeze: :today`)
+        def_node_matcher :has_metadata?, <<-PATTERN
+          (block
+            {
+              (send nil? #example_method? str _ ...)
+              (send nil? #example_method? !str ...)
+            }
+            ...
+          )
+        PATTERN
+
+        # Matchers examples with:
+        # - expectation statements exclusively
+        # - no metadata (e.g. `freeze: :today`)
+        # - no title (e.g. `it('jumps over the lazy dog')`)
+        # and skips `its` with an array argument due to ambiguous conversion
+        #   e.g. the SUT can be an object (`expect(object.property)`)
+        #   or a hash/array (`expect(hash['property'])`)
+        # and also skips matchers with known side-effects
+        def_node_matcher :examples_for_autocorrect?, <<-PATTERN
+          [
+            #example_without_metadata_and_expectations_solely?
+            !#example_has_title?
+            !#its_with_array_argument?
+            !#contains_heredoc?
+            !#example_with_side_effects?
+          ]
+        PATTERN
+
+        # Matchees the example with a title (e.g. `it('is valid')`)
+        def_node_matcher :example_has_title?, <<-PATTERN
+          (block
+            (send nil? #example_method? str ...)
+            ...
+          )
+        PATTERN
+
+        def_node_matcher :its_with_array_argument?, <<-PATTERN
+          (block (send nil? :its array) ...)
+        PATTERN
+
+        # Searches for HEREDOC in examples. It can be tricky to aggregate,
+        # especially when interleaved with parenthesis or curly braces.
+        def contains_heredoc?(node)
+          return unless node.is_a? AST::Node
```

Comment:
you can replace with `node.respond_to(:heredoc?) && node.heredoc?`

Reply:
Also if you use `each_child_node` instead of `children` you will always have an AST node